There is absolutely no point in using libtool files of libresid-builder, or even expect them to be present.

http://bugs.gentoo.org/416619
http://moc.daper.net/node/832

--- decoder_plugins/sidplay2/sidplay2.m4
+++ decoder_plugins/sidplay2/sidplay2.m4
@@ -19,10 +19,10 @@
 		if test "x$sidutils_OK" = "xyes"; then
 			s2lib=`$PKG_CONFIG --variable=libdir libsidplay2 2>/dev/null`
 			resid_OK="no"
-			AC_CHECK_FILE([$s2lib/libresid-builder.la],
-			              [resid_lib="$s2lib/libresid-builder.la"
+			AC_CHECK_FILE([$s2lib/libresid-builder.so],
+			              [resid_lib="$s2lib/libresid-builder.so"
 			               resid_OK="yes"],
-			              [resid_lib="$s2lib/sidplay/builders/libresid-builder.la"
+			              [resid_lib="$s2lib/sidplay/builders/libresid-builder.so"
 			               AC_CHECK_FILE($resid_lib, [resid_OK="yes"],)])
 			if test "x$resid_OK" = "xyes"; then
 				sidplay2_LDFLAGS="$resid_lib"
